packages/xchain-thorchain/src/client.ts

```diff
@@ -201,45 +206,107 @@ class Client implements ThorchainClient, XChainClient {
     }
   }
 
-  deposit = async ({ asset, amount, recipient, memo }: TxParams): Promise<TxHash> => {
-    return this.transfer({ asset, amount, recipient, memo })
+  private getDefaultStdTxFee = (): StdTxFee => {
+    return {
+      amount: [],
+      gas: '10000000',
+    }
+  }
+
+  deposit = async ({ asset, amount, memo }: DepositParam): Promise<TxHash> => {
+    try {
+      this.registerCodecs()
+
+      if (!asset) asset = AssetRune
+
+      const assetBalance = await this.getBalance(this.getAddress(), asset)
+      const fee = await this.getFees()
+      if (assetBalance.length === 0 || assetBalance[0].amount.amount().lt(amount.amount().plus(fee.average.amount()))) {
+        throw new Error('insufficient funds')
+      }
+
+      const signer = this.getAddress()
+
+      const msg: Msg = [
+        MsgNativeTx.fromJSON({
+          coins: [
+            {
+              asset: getDenomWithChain(asset),
+              amount: amount.amount().toString(),
+            },
+          ],
+          memo,
+          signer,
+        }),
+      ]
+
+      const unsignedStdTx = StdTx.fromJSON({
+        msg,
+        fee: this.getDefaultStdTxFee(),
+        signatures: [],
+        memo: '',
+      })
+
+      const transferResult = await this.thorClient.signAndBroadcast(
+        unsignedStdTx,
+        this.getPrivateKey(),
+        AccAddress.fromBech32(signer),
+      )
+
+      if (!isBroadcastSuccess(transferResult)) {
+        throw new Error(`failed to broadcast transaction: ${transferResult.txhash}`)
+      }
+
+      return transferResult?.txhash || ''
+    } catch (error) {
+      return Promise.reject(error)
+    }
   }
 
   transfer = async ({ asset, amount, recipient, memo }: TxParams): Promise<TxHash> => {
     try {
       this.registerCodecs()
 
+      if (!asset) asset = AssetRune
+
+      const assetBalance = await this.getBalance(this.getAddress(), asset)
+      const fee = await this.getFees()
+      if (assetBalance.length === 0 || assetBalance[0].amount.amount().lt(amount.amount().plus(fee.average.amount()))) {
+        throw new Error('insufficient funds')
+      }
+
       const transferResult = await this.thorClient.transfer({
         privkey: this.getPrivateKey(),
         from: this.getAddress(),
         to: recipient,
         amount: amount.amount().toString(),
-        asset: getDenom(asset || AssetRune),
+        asset: getDenom(asset),
         memo,
+        fee: this.getDefaultStdTxFee(),
```

Comment:
Why `getDefaultStdTxFee()` if we use `const fee = await this.getFees()` (line 273) for validating balances needed to send transaction?